File: src/main/java/edu/metrostate/trackingsystem/infrastructure/utils/GsonConfig.java
package edu.metrostate.trackingsystem.infrastructure.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import edu.metrostate.trackingsystem.domain.models.Price;

public class GsonConfig {
    public static Gson getGson() {
        return new GsonBuilder()
                .registerTypeAdapter(Price.class, new PriceDeserializer()) // Register custom deserializer
                .create();
    }
}

File: src/main/kotlin/edu/metrostate/dealership/infrastructure/utils/GsonConfig.kt
package edu.metrostate.dealership.infrastructure.utils

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import edu.metrostate.dealership.domain.models.Price

object GsonConfig {
    @JvmStatic
    val gson: Gson
        get() = GsonBuilder()
            .registerTypeAdapter(Price::class.java, PriceDeserializer()) // Register custom deserializer
            .create()
}